src/main/java/org/folio/tlib/postgres/impl/PgCqlQueryImpl.java

```diff
@@ -0,0 +1,315 @@
+      default:
+        throw new IllegalArgumentException("Unsupported operator " + base + " for: "
+            + termNode.toCQL());
+    }
+  }
+
+  static String numberOp(CQLTermNode termNode) {
+    String base = termNode.getRelation().getBase();
+    switch (base) {
+      case "==":
+        return "=";
+      case "=":
+      case "<>":
+      case ">":
+      case "<":
+      case "<=":
+      case ">=":
+        return base;
+      default:
+        throw new IllegalArgumentException("Unsupported operator " + base + " for: "
+            + termNode.toCQL());
+    }
+  }
+
+  /**
+   * See if this is a CQL query with a existence check (NULL or NOT NULL).
+   * <p>Empty term makes "IS NULL" for CQL relation =, "IS NOT NULL" for CQL relation <>.
+   * </p>
+   * @param field CQL field.
+   * @param termNode term.
+   * @return SQL op for NULL; null if not a NULL check.
+   */
+  String handleNull(PgCqlField field, CQLTermNode termNode) {
+    if (!termNode.getTerm().isEmpty()) {
+      return null;
+    }
+    String base = termNode.getRelation().getBase();
+    switch (base) {
+      case "=":
+        return field.getColumn() + " IS NULL";
+      case "<>":
+        return field.getColumn() + " IS NOT NULL";
+      default:
+        return null;
+    }
+  }
+
+  String handleTypeUuid(PgCqlField field, CQLTermNode termNode) {
+    String s = handleNull(field, termNode);
+    if (s != null) {
+      return s;
+    }
+    UUID id = UUID.fromString(termNode.getTerm()); // so IllegalArgumentException is thrown
+    String pgTerm = "'" + id + "'";
+    String op = basicOp(termNode);
+    return field.getColumn() + op + pgTerm;
+  }
+
+  /**
+   * Convert CQL term to Postgres term - exact - without C style escapes in result.
+   * <p> Double backslash is converted to backslash. Postgres quotes (') are escaped.
+   * Otherwise things are passed through verbatim.
+   * </p>
+   * @param termNode termNode which includes term and relation.
+   * @return Postgres term without C style escapes.
+   */
+  static String cqlTermToPgTermExact(CQLTermNode termNode) {
+    String cqlTerm = termNode.getTerm();
+    StringBuilder pgTerm = new StringBuilder();
+    boolean backslash = false;
+    for (int i = 0; i < cqlTerm.length(); i++) {
+      char c = cqlTerm.charAt(i);
+      if (c == '\\' && backslash) {
+        backslash = false;
+      } else {
+        pgTerm.append(c);
+        if (c == '\'') {
+          pgTerm.append('\''); // important to avoid SQL injection
+        }
+        backslash = c == '\\';
+      }
+    }
+    return pgTerm.toString();
+  }
+
+  /**
+   * CQL full text term to Postgres term.
+   * @see <a href="https://www.postgresql.org/docs/13/sql-syntax-lexical.html">
+   *   String Constants section</a>
+   *
+   * <p>At this stage masking is unsupported and rejected.</p>
+   * @param termNode which includes term and relation.
+   * @return Postgres term.
+   */
+  static String cqlTermToPgTermFullText(CQLTermNode termNode) {
```

Comment:
We already have a well tested fulltext search CQL string to SQL string converter:
https://github.com/folio-org/raml-module-builder/blob/v33.0.2/cql2pgjson/src/main/java/org/folio/cql2pgjson/util/Cql2SqlUtil.java#L260-L422
https://github.com/folio-org/raml-module-builder/blob/v33.0.2/cql2pgjson/src/test/java/org/folio/cql2pgjson/util/Cql2SqlUtilTest.java#L171-L479